core/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt

```diff
@@ -19,15 +19,38 @@ public class PolymorphicModuleBuilder<in Base : Any> @PublishedApi internal cons
     private val baseClass: KClass<Base>,
     private val baseSerializer: KSerializer<Base>? = null
 ) {
-    private val subclasses: MutableList<Pair<KClass<out Base>, KSerializer<out Base>>> = mutableListOf()
+    private val subclasses: MutableMap<KClass<out Base>, KSerializer<out Base>> = mutableMapOf()
     private var defaultSerializerProvider: ((Base) -> SerializationStrategy<Base>?)? = null
     private var defaultDeserializerProvider: ((String?) -> DeserializationStrategy<Base>?)? = null
 
+
+    /**
+     * Registers the child serializers for the sealed [subclass] [serializer] in the resulting module under the [base class][Base].
+     */
+    public inline fun <reified T : Base> subclassesOf(): Unit =
+        subclassesOf(serializer<T>())
+
+
+    /**
+     * Registers the subclasses of the given class as subclasses of the outer class. This currently requires `baseClass`
+     * to be sealed.
+     */
+    public fun <T: Base> subclassesOf(serializer: KSerializer<T>) {
```

Comment:
After thinking about it for a while, I came to the idea that `subclassesOf` is a misleading name. It implies that we can extract all subclasses for an arbitrary class, which we can't do — and as we all know, not every person will be reading documentation to find out that we are able to handle only sealed classes. So more appropriate name would be `sealedSubclassesOf` or `subclassesOfSealed` or something in that direction.